runtime/tools/spec-builder/test/build-spec.spec.js

```diff
@@ -0,0 +1,37 @@
+const assert = require('assert');
+const fs = require('fs');
+const { buildSpec, emptyActualRawFiles } = require('./utils/spawn-child-process');
+const { isEqual } = require('lodash');
+
+
+before(async function() {
+    await emptyActualRawFiles();
+    await buildSpec('./test/json-files/customSpec.json', './test/actual-raw-files/customSpecRaw.json');
+    await buildSpec('./test/json-files/customSpec-balances.json', './test/actual-raw-files/customSpec-balances.json');
+    await buildSpec('./test/json-files/customSpec-code.json', './test/actual-raw-files/customSpecRaw-code.json');
+});
+
+describe('Tool generates raw files', function(){
```

Comment:
I think that it will be beneficial to add several Negative test cases, f.e when the file that is provided does not have the right structure f.e `.genesis` or `.genesis.runtime` or `.genesis.runtime.system` or even the json file may have the `.balances` object but not the inner `balances` array, or it may have the balances array but that array is empty. All of those examples are valid tests.